creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -176,13 +176,26 @@ class SnapbackSM {
         `stateMachineQueue Job Lock Extension Failed - ID ${jobId} - Error ${err}`
       )
     })
+
+    // Re-queue state machine job when the current job fails or succeeds
+    this.stateMachineQueue.on('global:error', (error) => {
+      this.logError(`stateMachineQueue Job Error - ${error}`)
+      this.log(`stateMachineQueue adding another job after error`)
+      this.stateMachineQueue.add({ startTime: Date.now() })
+    })
     this.stateMachineQueue.on('global:completed', (jobId, result) => {
       this.log(
         `stateMachineQueue Job Completed - ID ${jobId} - Result ${result}`
       )
+      this.log(
+        `stateMachineQueue adding another job after successful completion`
+      )
+      this.stateMachineQueue.add({ startTime: Date.now() })
     })
     this.stateMachineQueue.on('global:failed', (jobId, err) => {
       this.logError(`stateMachineQueue Job Failed - ID ${jobId} - Error ${err}`)
+      this.log(`stateMachineQueue adding another job after previous job failed`)
+      this.stateMachineQueue.add({ startTime: Date.now() })
     })
 
     // Add stalled event handler for all queues
```

Comment:
did you intentionally not re-add job if job stalls

Reply:
yeah because a stalled job could still be running and complete later, right? I think this issue will go away if we eventually run these jobs in a separate worker thread anyway

Reply:
agreed, glad you considered it